htdocs/core/lib/functions.lib.php

```diff
@@ -8110,14 +8110,15 @@ function dol_eval($s, $returnvalue = 0, $hideerrors = 1)
 	}
 
 	// We block using of php exec or php file functions
-	$forbiddenphpstrings = array("exec(", "passthru(", "shell_exec(", "system(", "proc_open(", "popen(", "eval(", "dol_eval(", "executeCLI(");
-	$forbiddenphpstrings = array_merge($forbiddenphpstrings, array("fopen(", "file_put_contents(", "fputs(", "fputscsv(", "fwrite(", "fpassthru(", "unlink(", "mkdir(", "rmdir(", "symlink(", "touch(", "umask("));
-	$forbiddenphpstrings = array_merge($forbiddenphpstrings, array('function(', '$$', 'call_user_func('));
-	$forbiddenphpstrings = array_merge($forbiddenphpstrings, array('_ENV', '_SESSION', '_COOKIE', '_GET', '_POST', '_REQUEST'));
-	$forbiddenphpregex = 'global\s+\$';
+	$forbiddenphpstrings = array(
+		"exec", "passthru", "shell_exec", "system", "proc_open", "popen", "eval", "dol_eval", "executeCLI",
+		"fopen", "file_put_contents", "fputs", "fputscsv", "fwrite", "fpassthru", "unlink", "mkdir", "rmdir",
+		"symlink", "touch", "umask", 'function', 'call_user_func',
+		'_ENV', '_SESSION', '_COOKIE', '_GET', '_POST', '_REQUEST'
+	);
+	$forbiddenphpregex = 'global\s+\$|\b(' . implode('|', $forbiddenphpstrings) . ')\b|\$\$';
```

Comment:
Adding the \b( is a good tips, but it breaks the test on using here _ENV where no ( must be added